.github/workflows/check-pull-request.yml

```diff
@@ -37,6 +37,24 @@ jobs:
         with:
           fetch-depth: 1
 
+      - name: Set up Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'adopt'
+          java-version: '17'
+          cache: 'gradle'
+
       - name: Run Tests using Gradle Wrapper
-        run: ./gradlew test --no-daemon
+        run: |
+          ./gradlew --no-daemon test jacocoTestReport -x assemble
+
+      - name: Publish coverage reports
+        uses: codecov/codecov-action@v4
+        with:
+          fail_ci_if_error: true 
+          files: ./bridge/build/reports/jacoco/test/jacocoTestReport.xml
+          name: frontend-library
+          token: ${{ secrets.CODECOV_TOKEN }}
+          verbose: true
+          plugin: noop
 
```

Comment:
Also, can you add flags in order to difference the coverage metrics?
https://github.com/marketplace/actions/codecov

Reply:
The [flags feature](https://about.codecov.io/product/feature/flags/) it's targeted mainly for mono-repo codebases. Or if it has different kind of tests, but we only have unit testing. I don't think is worth it
![image](https://github.com/Pending-Name-21/frontend-library/assets/64939916/51c6353f-b9a4-48d4-96ec-e9203ece59a1)

Reply:
Sure

Reply:
Nice

Reply:
I see you added our flag but I have a question, is the flag service generated by default?

Reply:
The [GitHub action](https://github.com/codecov/codecov-action) does that by default anyway you can read the [docs](https://docs.codecov.com/docs/flags)
![image](https://github.com/Pending-Name-21/frontend-library/assets/64939916/169c3b33-3f37-4004-aefb-89c9093b35f4)

Reply:
Okay!